main/src/main/scala/sbt/internal/RemoteCache.scala

```diff
@@ -0,0 +1,217 @@
+        Classpaths.getPublishTo(pushRemoteCacheTo.value).name,
+        ivyLoggingLevel.value,
+        isSnapshot.value
+      )
+    },
+    pullRemoteCache := {
+      val s = streams.value
+      val smi = scalaModuleInfo.value
+      val dr = (pullRemoteCache / dependencyResolution).value
+      val is = (pushRemoteCache / ivySbt).value
+      val t = crossTarget.value / "cache-download"
+      val p = remoteCacheProjectId.value
+      val id = remoteCacheId.value
+      val compileAf = (Compile / compileAnalysisFile).value
+      val compileOutput = (Compile / classDirectory).value
+      val testAf = (Test / compileAnalysisFile).value
+      val testOutput = (Test / classDirectory).value
+      val testStreams = (Test / test / streams).value
+      val testResult = Defaults.succeededFile(testStreams.cacheDirectory)
+
+      val deps = Vector(p.classifier(cachedCompileClassifier), p.classifier(cachedTestClasifier))
+      val mconfig = dummyModule(smi, deps)
+      val m = new is.Module(mconfig)
+      dr.retrieve(m, t, s.log) match {
+        case Right(xs0) =>
+          val xs = xs0.distinct
+          xs.find(_.toString.endsWith(s"$id-$cachedCompileClassifier.jar")) foreach { jar: File =>
+            extractCache(jar, compileOutput, compileAf, None)
+          }
+          xs.find(_.toString.endsWith(s"$id-$cachedTestClasifier.jar")) foreach { jar: File =>
+            extractCache(jar, testOutput, testAf, Some(testResult))
+          }
+          ()
+        case Left(unresolvedWarning) =>
+          s.log.info(s"remote cache not found for ${id}")
+      }
+    },
+    remoteCachePom := {
+      val s = streams.value
+      val config = makePomConfiguration.value
+      val publisher = Keys.publisher.value
+      publisher.makePomFile((pushRemoteCache / ivyModule).value, config, s.log)
+      config.file.get
+    },
+    remoteCachePom / packagedArtifact := ((makePom / artifact).value -> remoteCachePom.value),
+  ) ++ inTask(pushRemoteCache)(
+    Seq(
+      ivyConfiguration := {
+        val other = pushRemoteCacheTo.value.toVector
+        val config0 = Classpaths.mkIvyConfiguration.value
+        config0
+          .withOtherResolvers(other)
+          .withResolutionCacheDir(crossTarget.value / "alt-resolution")
+      },
+      ivySbt := {
+        val config0 = ivyConfiguration.value
+        Credentials.register(credentials.value, streams.value.log)
+        new IvySbt(config0, CustomHttp.okhttpClient.value)
+      },
+      ivyModule := {
+        val is = ivySbt.value
+        new is.Module(moduleSettings.value)
+      },
+      moduleSettings := {
+        val smi = scalaModuleInfo.value
+        ModuleDescriptorConfiguration(remoteCacheProjectId.value, projectInfo.value)
+          .withScalaModuleInfo(smi)
+      },
+      pushRemoteCache.in(Defaults.TaskZero) := (Def.task {
+        val s = streams.value
+        val config = pushRemoteCacheConfiguration.value
+        IvyActions.publish(ivyModule.value, config, s.log)
```

Comment:
can this override with any other filestore?

Reply:
https://www.seafile.com/home/

Reply:
> can this override with any other filestore?

Yes. In the blog post I used:

```scala
ThisBuild / pushRemoteCacheTo := Some(MavenCache("local-cache", file("/tmp/remote-cache")))
```